src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -1556,7 +1558,35 @@ public boolean hasAlly(Town town) {
 	public boolean hasMutualAlly(Town town) {
 		return hasAlly(town) && town.hasAlly(this);
 	}
+	/**
+	 * Only to be used when loading the database.
+	 * @param towns List&lt;Town&gt; which will be loaded in as trusted towns.
+	 */
+	public void loadTrustedTowns(List<Town> towns) {
+		for (Town trustTown : towns) {
+			trustedTowns.put(trustTown.getUUID(), trustTown);
+		}
+	}
+
+	public void addTrustedTown(Town town) {
+		trustedTowns.put(town.getUUID(), town);
+	}
+
+	public void removeTrustedTown(Town town) {
+		trustedTowns.remove(town.getUUID());
+	}
 
+	public boolean removeAllTrustedTowns() {
+		for (Town trusted : new ArrayList<>(getAllies())) {
```

Comment:
You don't want to be using getAllies() here.

Reply:
Oh can't believe I failed to notice this, will update this now and add/remove events in API.

Reply:
Have implemented this change alongside new events for API in latest commit.